src/main/java/br/com/sbk/sbking/core/HandEvaluations.java

```diff
@@ -90,4 +92,63 @@ public boolean isBalanced() {
         && this.getNumberOfDoubletonSuits() <= 1;
   }
 
+  public boolean hasEightOrMoreCardsInAnySuit() {
+    return hasNOrMoreCardsInLongestSuit(8);
+  }
+
+  public boolean hasSixCardsInLongestSuit() {
+    return hasNCardsInLongestSuit(6);
+  }
+
+  private boolean hasNCardsInLongestSuit(int n) {
+    int longestSuit = this.getNumberOfCardsInLongestSuit();
+    return longestSuit == n;
+  }
+
+  private boolean hasNOrMoreCardsInLongestSuit(int n) {
+    int longestSuit = this.getNumberOfCardsInLongestSuit();
+    return longestSuit >= n;
+  }
+
+  public boolean hasTwoOutOfThreeHigherCards() {
+    Predicate<Card> isThreeHigherCards = card -> card.getRank().equals(Rank.ACE) || card.getRank().equals(Rank.KING)
```

Comment:
@rulojuka please let  me know if you have an idea to test this method, because it is necessary to evaluate the rank of the card.

Reply:
Just as usual. You create some different cases of HandEvaluations with different cards in the Hand and test them separetely. Like `handWithKingAndQueenShouldHaveTwoOutOfThreeHigherCards`...

Reply:
As this is a really small predicate, you can inline it as a lambda function inside the filter at the next line. Also, you could do `card.getRank()` only once.
```java
.filter(card -> {
      Rank rank = card.getRank();
      return Rank.ACE.equals(rank) || Rank.KING.equals(rank) || Rank.QUEEN.equals(rank);
    })
```